src/inodes/INodeManager.ts

```diff
@@ -210,25 +207,11 @@ class INodeManager {
     ...inos: Array<INodeIndex>
   ): ResourceAcquire<DBTransaction> {
     return async () => {
-      const locksAcquire = this.locks.lock(
-        ...inos.map<[INodeIndex, typeof Lock]>((ino) => [ino, Lock]),
+      const [transactionRelease, tran] = await this.db.transaction()();
+      await tran!.lock(
+        ...inos.map<[INodeIndex, 'write']>((ino) => [ino, 'write']),
       );
-      const transactionAcquire = this.db.transaction();
-      const [locksRelease] = await locksAcquire();
-      let transactionRelease, tran;
-      try {
-        [transactionRelease, tran] = await transactionAcquire();
-      } catch (e) {
-        await locksRelease();
-        throw e;
-      }
-      return [
-        async () => {
-          await transactionRelease();
-          await locksRelease();
-        },
-        tran,
-      ];
+      return [transactionRelease, tran];
```

Comment:
Do we need this function anymore...?

Can this be changed to using `withF`... rather than `inodeMgr.transaction`.

Just wondering if we should preserve this old style of API.

Reply:
I can inline it, but it's a nice wrapper for locking the inodes along side the transaction.

Reply:
This is used in 5 different places in a weird way. it's not used with a `WithF` block in any of these places so a closer look is needed before removing this. Might not be worth the time right now.

Reply:
WE determined that this is fine as it is.